backend/api/src/api/v1/gql/chat.rs

```diff
@@ -0,0 +1,143 @@
+        }
+
+        let session = request_context.get_session(global).await?;
+
+        let user_id = session
+            .as_ref()
+            .ok_or_else(|| GqlError::Unauthorized.with_message("You need to be logged in"))?
+            .user_id;
+
+        let user = global
+            .user_by_id_loader
+            .load_one(user_id)
+            .await
+            .map_err_gql("Failed to fetch user")?
+            .ok_or(GqlError::InvalidInput.with_message("User not found"))?;
+
+        let _chat = global
+            .chat_room_by_id_loader
+            .load_one(chat_id)
+            .await
+            .map_err_gql("Failed to fetch chat")?
+            .ok_or(GqlError::InvalidInput.with_message("Chat not found"))?;
+
+        let db_message = sqlx::query_as!(
+            chat_message::Model,
+            "INSERT INTO chat_messages (chat_room_id, author_id, message, created_at) VALUES ($1, $2, $3, $4) RETURNING *",
+            chat_id,
+            user_id,
+            content,
+            Utc::now()
+        )
+        .fetch_one(&*global.db)
+        .await
+        .map_err_gql("Failed to send message")?;
+
+        let message_to_send = Message {
+            message_id: db_message.id,
+            chat_id,
+            username: user.username,
+            content: content.to_string(),
+            message_type: "message".to_string(),
+            metadata: HashMap::from([("color".to_string(), "teal".to_string())]),
+        };
+
+        match global
+            .redis_pool
+            .publish::<String, String, String>(
+                format!("chat:{}", chat_id),
+                serde_json::json!(message_to_send).to_string(),
+            )
+            .await
+        {
+            Ok(_) => {}
+            Err(_) => {
+                return Err(GqlError::InternalServerError.with_message("Failed to publish message"));
+            }
+        };
+
+        Ok(message_to_send)
+    }
+}
+
+#[Subscription]
+impl ChatSubscription {
+    // Listen to new messages in chat.
+    pub async fn new_message<'ctx>(
+        &self,
+        ctx: &'ctx Context<'_>,
+        #[graphql(desc = "Chat to subscribe to.")] chat_id: i64,
+    ) -> Result<impl Stream<Item = Message> + 'ctx> {
+        let global = ctx.get_global();
+
+        let welcome_message = Message {
+            message_id: 0,
+            chat_id,
+            username: "".to_string(),
+            metadata: HashMap::new(),
+            message_type: "info".to_string(),
+            content: "Connected to chat".to_string(),
+        };
+
+        let _chat = match global.chat_room_by_id_loader.load_one(chat_id).await {
+            Ok(Some(chat)) => chat,
+            Ok(None) => {
+                return Err(GqlError::InvalidInput.with_message("Chat not found"));
+            }
+            Err(_) => {
+                return Err(GqlError::InternalServerError.with_message("Failed to fetch chat"));
+            }
+        };
```

Comment:
Perhaps something like this? (untested + unformatted)

```suggestion
        match global.chat_room_by_id_loader.load_one(chat_id).await {
            Ok(Some(_)) => (),
            Ok(None) => {
                return Err(GqlError::InvalidInput.with_message("Chat not found"));
            }
            Err(err) => {
                tracing::error!("failed to query chat room: {}", err);
                return Err(GqlError::InternalServerError.with_message("Failed to fetch chat"));
            }
        };
```